app/controllers/users/two_factor_authentication_controller.rb

```diff
@@ -44,7 +47,7 @@ def backup_code_redirect
     end
 
     def redirect_on_nothing_enabled
-      redirect_to two_factor_options_url
+      redirect_to login_two_factor_options_path
```

Comment:
> FWIW I think this is what it should be today? It seems to get stuck in a redirect loop with the current logic, if none of the other redirects apply.

Okay, not quite, but the logic is a bit tricky. I think what existed here previously was assuming that "nothing enabled" means that a user hasn't set up 2FA, and was relied upon as part of the initial account setup to redirect to the MFA setup page. However, since this can now include scenarios which include users who have set up MFA already, a redirect loop happens because `two_factor_options_url` (the MFA setup path) guards against users who already have 2FA set up and redirects them _back_ to this controller:

https://github.com/18F/identity-idp/blob/dfec5fe3612df835979b5443604a95844f4082a5/app/controllers/concerns/mfa_setup_concern.rb#L4-L9

So maybe this needs to be updated to account for both scenarios, where MFA is and is not yet set up. Something like:

```rb
if MfaPolicy.new(current_user).two_factor_enabled?
  redirect_to login_two_factor_options_path
else
  redirect_to two_factor_options_url
end
```